router/admin.go

```diff
@@ -27,7 +35,217 @@ func (ra *RouterAdmin) Status() interface{} {
 		routerStatus["name"] = name
 		routerStatus["success-count"] = router.successCount
 		routerStatus["failure-count"] = router.failCount
+		routerFailedList := make([]string, 0)
+		router.failedEventsListMutex.RLock()
+		for e := router.failedEventsList.Front(); e != nil; e = e.Next() {
+			status, _ := json.Marshal(e.Value)
+			routerFailedList = append(routerFailedList, string(status))
+		}
+		router.failedEventsListMutex.RUnlock()
+		if len(routerFailedList) > 0 {
+			routerStatus["recent-failedstatuses"] = routerFailedList
+		}
 		statusList = append(statusList, routerStatus)
+
 	}
 	return statusList
 }
+
+type RouterRpcHandler struct {
+}
+
+type JobCountsByStateAndDestination struct {
+	Count       int
+	State       string
+	Destination string
+}
+
+type FailedStatusErrorCodeCountsByDestination struct {
+	Count       int
+	ErrorCode   string
+	Destination string
+}
+
+type JobCountByConnections struct {
+	Count         int
+	SourceId      string
+	DestinationId string
+}
+
+type JobStatusCountsByExecTime struct {
+	Count int
+	State string
+	Rank  int
+}
+
+type DSStats struct {
+	JobCountsByStateAndDestination           []JobCountsByStateAndDestination
+	FailedStatusErrorCodeCountsByDestination []FailedStatusErrorCodeCountsByDestination
+	JobCountByConnections                    []JobCountByConnections
+	JobStatusCountsByExecTime                []JobStatusCountsByExecTime
+	UnprocessedJobCounts                     int
+	err                                      error `json:"-"`
+}
+
+// group_by job_status
```

Comment:
Let's add some example outputs here? To be able to understand easily what each function does.

Reply:
added table format